cluster/src/main/java/com/hartwig/pipeline/cleanup/Cleanup.java

```diff
@@ -30,50 +32,55 @@
         this.arguments = arguments;
         this.dataproc = dataproc;
+        this.somaticMetadataApi = somaticMetadataApi;
     }
 
     public void run(SomaticRunMetadata metadata) {
         if (!arguments.cleanup()) {
             return;
         }
         LOGGER.info("Cleaning up all transient resources on complete somatic pipeline run (runtime buckets and dataproc jobs)");
-        String referenceSampleName = metadata.reference().sampleId();
-        String tumorSampleName = metadata.tumor().sampleId();
-        Run referenceRun = Run.from(referenceSampleName, arguments);
-        deleteBucket(referenceRun.id());
-        Run tumorRun = Run.from(tumorSampleName, arguments);
-        deleteBucket(tumorRun.id());
-        deleteBucket(Run.from(referenceSampleName, tumorSampleName, arguments).id());
 
-        try {
-            Dataproc.Projects.Regions.Jobs jobs = dataproc.projects().regions().jobs();
-            ListJobsResponse execute = jobs.list(arguments.project(), arguments.region()).execute();
-            for (Job job : execute.getJobs()) {
-                if (job.getReference().getJobId().startsWith(referenceRun.id()) || job.getReference()
-                        .getJobId()
-                        .startsWith(tumorRun.id())) {
-                    LOGGER.debug("Deleting complete job [{}]", job.getReference().getJobId());
-                    deleteJob(jobs, job);
+        deleteBucket(Run.from(metadata.reference().sampleId(), metadata.tumor().sampleId(), arguments).id());
+        cleanupSample(metadata.reference());
+        cleanupSample(metadata.tumor());
+    }
 
-                    Job existingJob = Failsafe.with(new RetryPolicy<>().handleResultIf(Objects::nonNull)
-                            .onFailedAttempt(objectExecutionCompletedEvent -> deleteJob(jobs, job))
-                            .withDelay(Duration.ofSeconds(RETRY_DELAY))
-                            .withMaxRetries(MAX_RETRIES)).get(getJobAndReturnNullOn404(jobs, job));
-                    if (existingJob != null) {
-                        LOGGER.warn("Job [{}] still exists after [{}] attempts to delete it. Aborting, it will need to be deleted by hand",
-                                job.getReference().getJobId(),
-                                MAX_RETRIES);
+    private void cleanupSample(final SingleSampleRunMetadata metadata) {
+        if (!somaticMetadataApi.hasDependencies(metadata.sampleName())) {
+            Run run = Run.from(metadata.sampleId(), arguments);
+            deleteBucket(run.id());
+            try {
+                Dataproc.Projects.Regions.Jobs jobs = dataproc.projects().regions().jobs();
+                ListJobsResponse execute = jobs.list(arguments.project(), arguments.region()).execute();
+                for (Job job : execute.getJobs()) {
+                    if (job.getReference().getJobId().startsWith(run.id())) {
+                        LOGGER.debug("Deleting complete job [{}]", job.getReference().getJobId());
```

Comment:
Feels like this might be more of an `INFO` message to me. And maybe the `ensureDeleted` could be called by the `deleteJob`?